include/k4a/k4a.hpp

```diff
@@ -870,10 +901,42 @@ class transformation
         }
     }
 
+    /** Transforms depth map and a custom image into the geometry of the color camera.
+     * Throws error on failure
+     *
+     * \sa k4a_transformation_depth_image_to_color_camera_custom
+     * Creates a new image with the output.
+     */
+    std::pair<image, image>
+    depth_image_to_color_camera_custom(const image &depth_image,
+                                       const image &custom_image,
+                                       k4a_transformation_interpolation_type_t interpolation_type,
+                                       uint32_t invalid_custom_value) const
+    {
+        image transformed_depth_image = image::create(K4A_IMAGE_FORMAT_DEPTH16,
+                                                      m_color_resolution.width,
+                                                      m_color_resolution.height,
+                                                      m_color_resolution.width *
+                                                          static_cast<int32_t>(sizeof(uint16_t)));
+        image transformed_custom_image = image::create(custom_image.get_format(),
+                                                       m_color_resolution.width,
+                                                       m_color_resolution.height,
+                                                       m_color_resolution.width * (custom_image.get_stride_bytes() /
```

Comment:
stride / width doesn't necessarily give you the pixel size. Stride can be larger than the width * pixels size to account for alignment concerns. or to subsection an image buffer. (a DEPTH16 image could be 5 pixels wide and have a stride of 16 bytes for example)

The only way to know the correct size would be to look at the pixel format.

Reply:
How is this? Are there any good other methods for that?
```cpp
const int32_t bytes_per_pixel = static_cast<int32_t>(
            (custom_image.get_format() == K4A_IMAGE_FORMAT_CUSTOM8) ? sizeof(int8_t) : sizeof(int16_t));
image transformed_custom_image = image::create(custom_image.get_format(),
                                               m_color_resolution.width,
                                               m_color_resolution.height,
                                               m_color_resolution.width * bytes_per_pixel);
```

Reply:
I think is ok, since the transformation function should fail if it is anything other than CUSTOM8 or CUSTOM16. The only risk is that if there was a future CUSTOM32 or some other type that the transformation function allows you to transform this code will default to 2 bytes per pixel which may be wrong. It should still be safe though, the worst that would happen is the function would fail or you would allocate extra memory.

In general I prefer to be more explicit and validate that custom_image.get_format() is one of the two exact types expected and fail otherwise. In this case you are adding logic that is only valid for those two formats out of a larger set of possibilities, so its safer to validate that the image is one of those two formats first rather then rely on later logic to ensure correctness.

Reply:
I see. It may be redundant, but I think this will work and able to prepare for future expansion.
```cpp
int32_t bytes_per_pixel;
switch (custom_image.get_format())
{
    case K4A_IMAGE_FORMAT_CUSTOM8:
        bytes_per_pixel = static_cast<int32_t>(sizeof(int8_t));
        break;
    case K4A_IMAGE_FORMAT_CUSTOM16:
        bytes_per_pixel = static_cast<int32_t>(sizeof(int16_t));
        break;
    default:
        throw error("Failed to support this format of custom image!");
}
image transformed_custom_image = image::create(custom_image.get_format(),
                                               m_color_resolution.width,
                                               m_color_resolution.height,
                                               m_color_resolution.width * bytes_per_pixel);
```

Reply:
I like it.

Maybe it would benefit us to add a k4a_get_pixel_size_from_image_format function for this sort of thing in the future, but for now this seems pretty clean.

Reply:
> Maybe it would benefit us to add a k4a_get_pixel_size_from_image_format function for this sort of thing in the future

I agree.